designs/2024-support-l10n-rule-messages/README.md

```diff
@@ -0,0 +1,185 @@
+
+   Example:
+
+   ```js
+   module.exports = {
+      meta: {
+         messages: {
+            someMessageId: {
+               en: "Default message in English",
+               es: "Mensaje predeterminado en español"
+            }
+         }
+      },
+      create(context) {
+         return {
+            Identifier(node) {
+               context.report({
+                  node,
+                  messageId: "someMessageId"
+               });
+            }
+         };
+      }
+   };
+   ```
+
+2. Support a way using `eslint.config.js` file to configure message info by message ID.
+   This way should allow users to provide some plugins to update other plugins' messages. This approach allows users to customize the messages for rules provided by plugins.
+
+   Example:
+    
+   ```js
+   module.exports = {
+     locale: "es",
+     rules: [/* ... */],
+     messages: {
+       "plugin/ruleId": {
+         someMessageId: "Mensaje predeterminado en español"
+       }
+     }
+   };
+   ```
+
+   and eslint will use the message info to replace the message in the rule if the message ID is found in the configuration.
+
+for corner cases, we need to consider the following:
+
+1. If a translation is missing for a specific language, ESLint should fall back to the default English message.
+2. ESLint should support original flat rules structure and the new structure with localized messages.
+
+## Documentation
+
+<!--
+    How will this RFC be documented? Does it need a formal announcement
+    on the ESLint blog to explain the motivation?
+-->
+
+1. Update the official ESLint documentation to include information on how to create and use localization files.
+2. Provide examples and guidelines for contributing translations, including third party integrations.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think 
+    about any opposing viewpoints that reviewers might bring up. 
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+1. Increased Maintenance Burden: Adding support for localization will increase the complexity of the ESLint codebase. Maintaining translations and ensuring they are up-to-date with rule changes will require additional effort from both the core team and contributors.
+2. It's better to use a module to handle the localization such as typescript using `@types/some-module` but it's not possible to use a module to handle the localization in ESLint.
```

Comment:
What part isn't possible?

Reply:
We need to scan the entire node_modules directory to locate the `@i18n/module` for each `eslint-plugin-module` and implement a mechanism to load the `@i18n/module`. Does this approach seem reasonable?"

Reply:
Ah I see. Yes, we don't want ESLint to scan the `node_modules` directory (if ESLint isn't running in Node.js, that directory might not exist). We'd need people to manually import the module into their config file.